cmd/minishift/cmd/root.go

```diff
@@ -213,14 +217,38 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If *minishift profile set* or *minishift start --profile* command is used.
 	for i, arg := range os.Args {
+		// This will match if `--profile` flag is used
 		if arg == "--"+profileFlag {
+			// This make sure if user specify *profile* flag then
+			// it should not panic with out of index error
+			if len(os.Args) <= i+1 {
+				break
+			}
 			profileName = os.Args[i+1]
 			if !cmdUtil.IsValidProfileName(profileName) {
 				atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
 			}
 			break
 		}
+		// This will match if we used *profile* command
+		if arg == profileFlag {
```

Comment:
This would introduce bugs for the aliases for `minishift profile` command i.e. `instance` and `profiles`.